// cote/dist/prom-client.js
const client = require('prom-client')

class Gauge {
    static getInstance(){
        if(!this.instance){
            let instance = new client.Gauge({
                name: 'discovery_services',
                help: 'Monitoring services',
                labelNames: ['service'] 
            })
            return instance
        }
        return Gauge.instance
    }
}
class CountSuccessServices {
    static getInstance(){
        if(!this.instance){
            let instance = new client.Gauge({
                name: 'total_success_request_servies',
                help: 'Counting total request in services',
                labelNames: ['service', 'type'] 
            })
            return instance
        }
        return Gauge.instance
    }
}
class CountFailServices {
    static getInstance(){
        if(!this.instance){
            let instance = new client.Gauge({
                name: 'total_failed_request_servies',
                help: 'Counting total request in services',
                labelNames: ['service', 'type'] 
            })
            return instance
        }
        return Gauge.instance
    }
}

module.exports = {
    Gauge: Gauge.getInstance(),
    CountSuccessServices: CountSuccessServices.getInstance(),
    CountFailServices: CountFailServices.getInstance()
}
